jetty-start/src/main/java/org/eclipse/jetty/start/CommandLineBuilder.java

```diff
@@ -54,86 +55,204 @@ public static String findJavaBin()
+                    builder.append("'\\''");
+            }
+            else
+                builder.append(c);
+        }
+        builder.append('\'');
+
+        return builder.toString();
+    }
+
+    /**
+     * Add a simple argument to the command line, quoted if necessary.
      *
      * @param arg the simple argument to add
      */
     public void addArg(String arg)
     {
         if (arg != null)
         {
+            if (commandLine.length() > 0)
+                commandLine.append(separator);
             args.add(arg);
+            commandLine.append(shellQuoteIfNeeded(arg));
         }
     }
 
     /**
-     * Similar to {@link #addArg(String)} but concats both name + value with an "=" sign, quoting were needed, and excluding the "=" portion if the value is
-     * undefined or empty.
-     *
-     * <pre>
-     *   addEqualsArg("-Dname", "value") = "-Dname=value"
-     *   addEqualsArg("-Djetty.home", "/opt/company inc/jetty (7)/") = "-Djetty.home=/opt/company\ inc/jetty\ (7)/"
-     *   addEqualsArg("-Djenkins.workspace", "/opt/workspaces/jetty jdk7/") = "-Djenkins.workspace=/opt/workspaces/jetty\ jdk7/"
-     *   addEqualsArg("-Dstress", null) = "-Dstress"
-     *   addEqualsArg("-Dstress", "") = "-Dstress"
-     * </pre>
-     *
+     * @deprecated use {@link #addArg(String, String)}
+     */
+    @Deprecated
+    public void addEqualsArg(String name, String value)
+    {
+        addArg(name, value);
+    }
+
+    /**
+     * Add a "name=value" style argument to the command line with
+     * name and value quoted if necessary.
      * @param name the name
      * @param value the value
      */
-    public void addEqualsArg(String name, String value)
+    public void addArg(String name, String value)
     {
+        Objects.requireNonNull(name);
+
+        if (commandLine.length() > 0)
+            commandLine.append(separator);
+
         if ((value != null) && (value.length() > 0))
         {
             args.add(name + "=" + value);
+            commandLine.append(shellQuoteIfNeeded(name)).append('=').append(shellQuoteIfNeeded(value));
         }
         else
         {
             args.add(name);
+            commandLine.append(shellQuoteIfNeeded(name));
         }
     }
 
     /**
-     * Add a simple argument to the command line.
-     * <p>
-     * Will <b>NOT</b> quote/escape arguments that have a space in them.
-     *
-     * @param arg the simple argument to add
+     * @deprecated use {@link #addArg(String)}
      */
+    @Deprecated
     public void addRawArg(String arg)
     {
-        if (arg != null)
+        addArg(arg);
+    }
+
+    /**
+     * Add a "-Oname" style option to the command line with no quoting.
+     * @param option the option
+     */
+    public void addOption(String option)
+    {
+        Objects.requireNonNull(option);
+        if (commandLine.length() > 0)
+            commandLine.append(separator);
+        commandLine.append(option);
+        args.add(option);
```

Comment:
To avoid duplication:

```suggestion
        addOption(option, null, null);
```